src/async_fuse/memfs/kv_engine/etcd_impl.rs

```diff
@@ -292,6 +616,102 @@ impl MetaTxn for EtcdTxn {
         self.buffer.insert(key, None);
     }
 
+    /// Try to campaign the master with simple txn
+    /// Old(reference to etcd client v3):
+    /// Create a key with prefix, and compare key (sorted by revision) with the key created by the same session.
+    /// If current key is the smallest, then the session is the master, and campaign success.
+    ///
+    /// New(For datenlord cache scenario):
+    /// A small batch of nodes in cluster, we just need a simple txn to get the master key.
+    /// The thundering herd is not the main problem, which will cause a bunch of etcd raft logs.
+    /// Use String as val can be easily compared.
+    async fn campaign(
+        &self,
+        key: &KeyType,
+        val: String,
+        lease_id: i64,
+    ) -> DatenLordResult<(bool, Option<String>)> {
```

Comment:
No need to use `Option<String>` , just `String` on normal case, otherwise should be treeted as `Err()`

Reply:
Ok, so we just need to treat `campaign failed` as Err.